Support BIGINT Primary Keys

labkey-matthewb opened the pull request:
#### Rationale
Implement support for tables with BIGINT primary keys.  E.g. RowId/ObjectId BIGSERIAL
Because of the large number of shared classes, utilities, services, etc, a lot of code was migrated to used Long.  Code should now assume that an generic integer object key (as opposed to a known table PK), could be a Long.

This PR is aimed at a) Supporting ObjectID BIGSERIAL b) making it possible (but not automatic) to migrate tables that want to use RowId BIGSERIAL.

#### Related Pull Requests
- <!-- list of links to related pull requests (replace this comment) -->

#### Changes
New safer collection classes
  class Int/LongHashMap
  class Int/LongHashSet
explicit Class for key values in Map Objects returned by BaseSelector
  Selector.getValueMap(Class key)
safe "cast" using asInteger or asLong, as opposed to (Integer) and (Long)

<!-- list of standard tasks (remove this comment to enable)
#### Tasks 📍
- [ ] Manual Testing
- [ ] Needs Automation
- [ ] Verify Fix
-->

labkey-matthewb committed:
checkpoint - migrate to using long/Long to represent any integer ID value (rowId, objectId, etc).

labkey-matthewb committed:
missed a repo

labkey-matthewb committed:
mock fix

labkey-matthewb committed:
Merge remote-tracking branch 'origin/develop' into fb_long_is_the_new_int

labkey-matthewb committed:
collections -> collections4

labkey-matthewb committed:
fix cast

labkey-matthewb committed:
getValueMap() -> getValueMap(Class)

labkey-matthewb committed:
go all in on IntHashMap and LongHashMap

labkey-matthewb committed:
Nab Integer->Long

labkey-matthewb committed:
Merge remote-tracking branch 'origin/develop' into fb_long_is_the_new_int

labkey-matthewb committed:
mismatched getter/setter

labkey-matthewb committed:
remove (Integer) casts
bulk search/replace

labkey-matthewb committed:
Merge remote-tracking branch 'origin/develop' into fb_long_is_the_new_int

labkey-adam committed:
Update all ObjectIds to BIGINT

labkey-nicka commented on viability/resources/schemas/dbscripts/postgresql/viability-25.000-25.001.sql:
```diff
@@ -0,0 +1 @@
+MiALTER TABLE viability.Results ALTER COLUMN ObjectId TYPE BIGINT;
```
Seems like a typo. Should be `ALTER` yea?

labkey-adam replied:
Dang... typed into wrong window or something

labkey-adam committed:
Fix typo

labkey-matthewb committed:
(Integer)

labkey-adam committed:
Bump module schema version so protein FK fix script actually runs

labkey-matthewb committed:
(Integer)

labkey-matthewb committed:
(Integer)

labkey-adam committed:
Eliminate Elispot Integer cast. Log exceptions.

labkey-matthewb committed:
LongHashSet structural search/replace

labkey-matthewb committed:
asInteger()

labkey-matthewb committed:
instanceof Integer
probably need a new helpers like
isIntegerObject() // instanceof Integer or Long
asIntegerOrNull() // null if not isIntegerObject()
asLongOrNull()    // null if not isIntegerObject()
